#                             ===========================================
#                             HOW TO DEFINE INPUT TOPICS FOR flowcean_ros
#                             ===========================================
#
# Structure:
# input_topics:
#   "<topic_name>":
#     - single_msg: <true/false>  # Optional, specify if the topic only contains a single message to grab
#     - include:  # Optional, specify fields to include
#       - "<field_name>"
#     - exclude:  # Optional, specify fields to exclude
#       - "<field_name>"
#     - qos_profile: "<profile_name>"  # Optional, specify QoS profile
#     - threshold: <value>  # Optional, specify individual threshold for the topic to be considered valid
#
# Note:
# -> Msg types are detected automatically based on the topic name.
# -> If include and exclude are not specified, flowcean_ros will subscribe to the entire message.
# -> Use either include or exclude. If both are specified, flowcean_ros prioritizes include.
#
#                             ===========================================
#                             HOW TO DEFINE OUTPUT TOPIC FOR flowcean_ros
#                             ===========================================
#
# Structure:
# output_topic:
#   - name: "<output_topic_name>"  # Name of the output topic
#   - msg_type: "<message_type>"  # Type of the message to be published
#   - qos_profile: "<profile_name>"  # QoS profile for the output topic

input_topics:
  "/amcl_pose":
    include:
      - "pose.pose.position.x"
      - "pose.pose.position.y"
      - "pose.pose.orientation.x"
      - "pose.pose.orientation.y"
      - "pose.pose.orientation.z"
      - "pose.pose.orientation.w"
    qos_profile: "SENSOR_DATA"
    threshold: 1.0
  # "/momo/pose":
  #   include:
  #     - "pose.position.x"
  #     - "pose.position.y"
  #     - "pose.orientation.x"
  #     - "pose.orientation.y"
  #     - "pose.orientation.z"
  #     - "pose.orientation.w"
  #   qos_profile: "SENSOR_DATA"
  #   threshold: 1.0
  "/scan":
    include:
      - "ranges"
      - "angle_min"
      - "angle_max"
      - "angle_increment"
      - "range_min"
      - "range_max"
    qos_profile: "SENSOR_DATA"
    threshold: 1.0
  "/map":
    include:
      - "data"
      - "info.resolution"
      - "info.width"
      - "info.height"
      - "info.origin.position.x"
      - "info.origin.position.y"
      - "info.origin.position.z"
      - "info.origin.orientation.x"
      - "info.origin.orientation.y"
      - "info.origin.orientation.z"
      - "info.origin.orientation.w"
    qos_profile: "SENSOR_DATA"
    threshold: 1.0
    one_time: True
  # "/delocalizations":
  #   include:
  #     - "data"
  "/particle_cloud":
    include:
      - "particles"
    qos_profile: "SENSOR_DATA"
    threshold: 1.0
  # "/position_error":
  #   include:
  #     - "data"
  # "/heading_error":
  #   include:
  #     - "data"
    
output_topics:
  "/output":
    msg_type: "std_msgs.msg.String"
    qos_profile: "SENSOR_DATA"

# "/amcl_pose":
#   msg_type: "geometry_msgs.msg.PoseWithCovarianceStamped"
#   fields:
#     - "pose.pose.position.x"
#     - "pose.pose.position.y"
#     - "pose.pose.orientation.x"
#     - "pose.pose.orientation.y"
#     - "pose.pose.orientation.z"
#     - "pose.pose.orientation.w"
#   custom:
#     qos_profile: "SENSOR_DATA"
#     threshold: 1.0

# "/scan":
#   msg_type: "sensor_msgs.msg.LaserScan"
#   fields:
#     - "ranges"
#     - "angle_min"
#     - "angle_max"
#     - "angle_increment"
#     - "range_min"
#     - "range_max"
#   custom:
#     qos_profile: "SENSOR_DATA"
#     threshold: 1.0

# "/map":
#   msg_type: "nav_msgs.msg.OccupancyGrid"
#   fields:
#     - "data"
#     - "info.resolution"
#     - "info.width"
#     - "info.height"
#     - "info.origin.position.x"
#     - "info.origin.position.y"
#     - "info.origin.position.z"
#     - "info.origin.orientation.x"
#     - "info.origin.orientation.y"
#     - "info.origin.orientation.z"
#     - "info.origin.orientation.w"
#   custom:
#     qos_profile: "SENSOR_DATA"
#     threshold: 1.0
#   one_time_data: true

# "/particle_cloud":
#   msg_type: "nav2_msgs.msg.ParticleCloud"
#   fields:
#     - "particles"
#   custom:
#     qos_profile: "SENSOR_DATA"
#     threshold: 1.0

